File: src/gameShop/ShopImpl.java
package gameShop;

import java.util.HashSet;
import java.util.Map;
import java.util.Set;

import item.FoodType;
import item.Seed;
import item.SeedImpl;
import item.SeedType;

public class ShopImpl implements Shop {

    private final Set<SeedType> seedItemList = new HashSet<>();
    private final Set<FoodType> foodItemList = new HashSet<>();

    public ShopImpl() {
        /* carico la set di Food(enum) */
        for (FoodType f : FoodType.values()) {
            foodItemList.add(f);
        }
        /* carico la set di SeedType(enum) */
        for (SeedType st : SeedType.values()) {
            seedItemList.add(st);

        }
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Set<SeedType> getSeedItemList() {
        return seedItemList;
    }

    /**
     *{@inheritDoc}
     */
    @Override
    public Set<FoodType> getFoodItemList() {
        return foodItemList;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public double sell(final FoodType fd) {
        return fd.getPrice();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public double sellAll(final Map<FoodType, Integer> map) {
        double somma = 0;
        for (FoodType f : map.keySet()) {
            for (int i = 0; i < map.get(f); i++) {
                somma += f.getPrice();
            }
        }
        return somma;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Seed buy(final SeedType st) {
        return new SeedImpl((SeedType) st);
    }
}

File: C#/RegaSimone/FarmingSimulator/FarmingSimulator/ShopImpl.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FarmingSimulator
{
    class ShopImpl : Shop
    {
        private HashSet<SeedType> seedItemList = new HashSet<SeedType>();
        private HashSet<FoodType> foodItemList = new HashSet<FoodType>();

        public ShopImpl()
        {
            /* carico la set di Food(enum) */
            foreach (FoodType f in FoodTypeMethods.GetValues())
            {
                foodItemList.Add(f);
            }
            /* carico la set di SeedType(enum) */
            foreach (SeedType st in SeedTypeMethods.GetValues())
            {
                seedItemList.Add(st);

            }
        }
        public Seed buy(SeedType st)
        {
            throw new NotImplementedException();
        }

        public HashSet<FoodType> GetFoodItemList()
        {
            return this.foodItemList;
        }

        public HashSet<SeedType> GetSeedItemList()
        {
            return this.seedItemList;
        }

        public double sell(FoodType f)
        {
            return FoodTypeMethods.GetPrice(f);
        }

        public double sellAll(Dictionary<FoodType, int> foodToSell)
        {
            double somma = 0;
            foreach(FoodType f in foodToSell.Keys)
            {
                for(int i=0; i< foodToSell[f]; i++)
                {
                    somma += FoodTypeMethods.GetPrice(f);
                }
            }
            return somma;
        }
    }
}
